hazelcast/src/main/java/com/hazelcast/jet/impl/JobExecutionService.java

```diff
@@ -316,27 +315,24 @@ public CompletableFuture<RawJobMetrics> runLightJob(
      *     init execution is retried.
      * </li></ul>
      */
-    public void initExecution(
+    public CompletableFuture<Void> initExecution(
             long jobId, long executionId, Address coordinator, int coordinatorMemberListVersion,
             Set<MemberInfo> participants, ExecutionPlan plan
     ) {
         ExecutionContext execCtx = addExecutionContext(
                 jobId, executionId, coordinator, coordinatorMemberListVersion, participants);
 
-        try {
-            jobClassloaderService.prepareProcessorClassLoaders(jobId);
-            Set<Address> addresses = participants.stream().map(MemberInfo::getAddress).collect(toSet());
-            ClassLoader jobCl = jobClassloaderService.getClassLoader(jobId);
-            doWithClassLoader(jobCl, () -> execCtx.initialize(coordinator, addresses, plan));
-        } finally {
-            jobClassloaderService.clearProcessorClassLoaders();
-        }
-
-
-        // initial log entry with all of jobId, jobName, executionId
-        logger.info("Execution plan for jobId=" + idToString(jobId)
-                + ", jobName=" + (execCtx.jobName() != null ? '\'' + execCtx.jobName() + '\'' : "null")
-                + ", executionId=" + idToString(executionId) + " initialized");
+        jobClassloaderService.prepareProcessorClassLoaders(jobId);
+        Set<Address> addresses = participants.stream().map(MemberInfo::getAddress).collect(toSet());
+        ClassLoader jobCl = jobClassloaderService.getClassLoader(jobId);
+        return doWithClassLoader(jobCl, () -> execCtx.initialize(coordinator, addresses, plan))
```

Comment:
This is not correct. It sets thread-local values, but then possibly executes on another thread.

Reply:
After analyzing I see what you mean. However, I think the doWithClassLoader here is not necessary, as ExecutionPlan.initProcSuppliers and others already wraps every processor creation in the doWithClassLoader. This one seems to be made "just in case", IMHO it's ok to remove it. Do you agree?

Reply:
The comment was about `jobClassloaderService.prepareProcessorClassLoaders()`.

Reply:
Oh, you are right! However, the comment is similar - we don't use deserializeWithCustomClassLoader, so we don't need this call. I removed it in the next commit.